### PR TITLE
Enable 'this.' Rule

### PR DESCRIPTION
**Pull request by garyrussell**
Gradle task currently only fixes one violation per line; it's not
too hard to fix but it's easy enough to run multiple times.

Pass 1

Manually Fix Inner Classes

Pass 2

Pass 3

Pass 4

Pass 5

Manually fix up inner classes.

### CODE REVIEW

**Comment by artembilan** [comment]
According to the Travis report, looks like you have enable not only `this.` rule :smile:

File: build.gradle [hunk 1]
```diff
@@ -177,6 +178,7 @@ subprojects { subproject ->
 
 	checkstyle {
 		configFile = new File(rootDir, "src/checkstyle/checkstyle.xml")
+		// ignoreFailures = true
```
Comment (artembilan) [inline build.gradle]: I found that we don't need this option.
The `--continue` does the trick.

File: src/checkstyle/fixThis.gradle [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+					// println line
+					ln++
+					def matcher = line =~ /Copyright (20\d\d)(?:-20\d\d)?/
+					if (matcher.count) {
+						years = matcher[0][1]
+						if (years != now) {
+							years = years + "-$now"
+						}
+						line.replaceFirst(/\{year}/, years)
+					}
```
Comment (artembilan) [inline src/checkstyle/fixThis.gradle]: I think the `if` above is a redundant copy/paste artifact.

Comment (garyrussell) [inline src/checkstyle/fixThis.gradle]: ?? Don't we want to change the copyright year if necessary?

**Comment by garyrussell** [comment]
Rebased, pushed.

File: src/checkstyle/fixThis.gradle (hunk 2 above)
Comment (artembilan) [inline src/checkstyle/fixThis.gradle]: Yes! Of course!
OK. Does it work really then?
How is that `{year}` in the line?

Comment (garyrussell) [inline src/checkstyle/fixThis.gradle]: No idea - I don't understand groovy :smile:

Comment (artembilan) [inline src/checkstyle/fixThis.gradle]: Yeah... That was a placeholder in my `fixHeaders` script in the header template.
I think the correct replacement here should be like this:
```
.replaceFirst(/\\E20\\d\\d\(\\-20\\d\\d\)\?\\Q/, years)
```
See `fixHeaders.gradle` for more info:
```
headerTemplate = new File('src/checkstyle/checkstyle-header.txt')
			.text
			.replaceAll(/\^\\Q|\\E\$|\^\$|\^\.\*\$/, '')
			.replaceFirst(/\\E20\\d\\d\(\\-20\\d\\d\)\?\\Q/, '{year}')
			.trim() + System.lineSeparator()  + System.lineSeparator()
```

:wink:

Comment (artembilan) [inline src/checkstyle/fixThis.gradle]: Doh.
This one:
```
.replaceFirst(/(20\d\d)(?:-20\d\d)?/, years)
```

Comment (garyrussell) [inline src/checkstyle/fixThis.gradle]: I'll try that - thanks - I was surprised there were no changes - I just assumed your previous one to fix the licenses touched all the same files.

Comment (garyrussell) [inline src/checkstyle/fixThis.gradle]: I ran with that and it made no changes, so let's just update the script during merge.

Thanks.

Comment (artembilan) [inline src/checkstyle/fixThis.gradle]: Yep! 
Merging...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/6d9afdfa0b15ca3ed68c26476143030be0123d25

**Pull request closed without merging**